tests/backup/backup_sanity_test.go

```diff
@@ -365,3 +366,154 @@ var _ = Describe("{BasicBackupCreation}", Label(TestCaseLabelsMap[BasicBackupCre
+				err = CreateBackupLocation(provider, backupLocationName, backupLocationUID, cloudCredName, cloudCredUID, getGlobalBucketName(provider), BackupOrgID, "", true)
+				dash.VerifyFatal(err, nil, "Creating backup location")
+			}
+		})
+
+		Step("Registering cluster for backup", func() {
+			log.InfoD("Registering cluster for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			err = CreateApplicationClusters(BackupOrgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+
+			clusterStatus, err := Inst().Backup.GetClusterStatus(BackupOrgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+
+			sourceClusterUid, err = Inst().Backup.GetClusterUID(ctx, BackupOrgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+
+			clusterStatus, err = Inst().Backup.GetClusterStatus(BackupOrgID, DestinationClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", DestinationClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", DestinationClusterName))
+		})
+
+		Step("Validate backup with partial success by failing CSI volumes", func() {
+			log.InfoD("Failing volumes with CSI backup type")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			k8sCore := core.Instance()
+			labelSelector := make(map[string]string)
+			labelSelector["backupVolumeType"] = "csi"
+			pvcList, err := k8sCore.GetPersistentVolumeClaims(namespace, labelSelector)
+			log.FailOnError(err, "Getting PVC list with provisioner based label selector")
+			if len(pvcList.Items) == 0 {
+				log.Fatalf("No PVCs found with the specified label selector")
+			}
+			scProvisioner, err = k8sCore.GetStorageProvisionerForPVC(&pvcList.Items[0])
+			dash.VerifyFatal(err, nil, " get GetStorageProvisionerForPVC")
+			log.Infof("The provisioner is %v", scProvisioner)
+			snapShotClassName := fmt.Sprintf("invalid-snapshotclass-%s", RandomString(3))
+			// Convert pvcList to a slice of PVC names
+			pvcNames := make([]string, len(pvcList.Items))
+			for i, pvc := range pvcList.Items {
+				pvcNames[i] = pvc.Name
+			}
+			_, err = CreateInvalidVolumeSnapshotClass(snapShotClassName, scProvisioner)
+			backupName := fmt.Sprintf("%s-%s-%v", "autogenerated-backup", namespace, RandomString(6))
+			err = CreateBackupWithPartialSuccessValidation(ctx, backupName, SourceClusterName, backupLocationName, backupLocationUID, scheduledAppContexts, labelSelectors, BackupOrgID, sourceClusterUid, "", "", "", "", pvcNames)
```

Comment:
@sabrarhussaini For this backup to fail, we should have passed invalid csi to backup creation request. can you confirm backup is actually ending up as partial backup.

Reply:
Correct Adil, but the backup was going into Partial Backup.
![image](https://github.com/portworx/torpedo/assets/105486766/b6e5c41a-aaa7-4a42-8390-24adb97a8a78)
I relied on just the backup status without paying attention to the testcase